lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -203,6 +206,11 @@ def record_supportability_metrics total_count, captured_count, dropped_count
         NewRelic::Agent.increment_metric(SEEN_METRIC, total_count)
         NewRelic::Agent.increment_metric(SENT_METRIC, captured_count)
       end
+
+      def truncate_message(message)
+        return if message.bytesize <= MAX_BYTES
+        message.byteslice(0...MAX_BYTES)
```

Comment:
(not a trick question - I don't know the answer) What does `#byteslice` give us here over `#truncate` (`message.truncate(MAX_BYTES)`)?

Reply:
Great question! Truncate is a Rails-created string extension. Byteslice is pure Ruby.

Reply:
Ah! Thank you. Ok, well then we should definitely use a pure Ruby approach. But now that I know that `#truncate` is an ActiveSupport-ism, I took a look at `#truncate`'s source and found that it does `string[0, MAX]` instead of our `0...MAX` approach. Now I want test the two approaches! But I won't let that hold up approval.

Reply:
Nice! I hesitate to use that approach because it relies on character length rather than bytesize. Though the two are often the same, the true limit is on the number of bytes, so I'd rather lean on specificity unless there's a big performance difference.

https://ruby-doc.org/core-3.1.1/String.html#method-i-length